commit 9d02e5d39d7b5e2806a5d98bdde24f4584e70528
Author: David Zafman <dzafman@redhat.com>
Date:   Mon Aug 5 20:57:48 2019 -0700

    osd mon mgr: Convert all network ping time output to milliseconds
    
    To output milliseconds (usec / 1000), treat as fixed point integers
    
    Signed-off-by: David Zafman <dzafman@redhat.com>

diff --git a/src/mgr/ClusterState.cc b/src/mgr/ClusterState.cc
index 0bb06229bd9..80eb6828a8d 100644
--- a/src/mgr/ClusterState.cc
+++ b/src/mgr/ClusterState.cc
@@ -339,21 +339,21 @@ bool ClusterState::asok_command(std::string_view admin_command, const cmdmap_t&
       f->dump_int("to osd", sitem.to);
       f->dump_string("interface", (sitem.back ? "back" : "front"));
       f->open_object_section("average");
-      f->dump_unsigned("1min", sitem.times[0]);
-      f->dump_unsigned("5min", sitem.times[1]);
-      f->dump_unsigned("15min", sitem.times[2]);
+      f->dump_format_unquoted("1min", "%s", fixed_u_to_string(sitem.times[0],3).c_str());
+      f->dump_format_unquoted("5min", "%s", fixed_u_to_string(sitem.times[1],3).c_str());
+      f->dump_format_unquoted("15min", "%s", fixed_u_to_string(sitem.times[2],3).c_str());
       f->close_section(); // average
       f->open_object_section("min");
-      f->dump_unsigned("1min", sitem.min[0]);
-      f->dump_unsigned("5min", sitem.min[1]);
-      f->dump_unsigned("15min", sitem.min[2]);
+      f->dump_format_unquoted("1min", "%s", fixed_u_to_string(sitem.min[0],3).c_str());
+      f->dump_format_unquoted("5min", "%s", fixed_u_to_string(sitem.min[1],3).c_str());
+      f->dump_format_unquoted("15min", "%s", fixed_u_to_string(sitem.min[2],3).c_str());
       f->close_section(); // min
       f->open_object_section("max");
-      f->dump_unsigned("1min", sitem.max[0]);
-      f->dump_unsigned("5min", sitem.max[1]);
-      f->dump_unsigned("15min", sitem.max[2]);
+      f->dump_format_unquoted("1min", "%s", fixed_u_to_string(sitem.max[0],3).c_str());
+      f->dump_format_unquoted("5min", "%s", fixed_u_to_string(sitem.max[1],3).c_str());
+      f->dump_format_unquoted("15min", "%s", fixed_u_to_string(sitem.max[2],3).c_str());
       f->close_section(); // max
-      f->dump_unsigned("last", sitem.last);
+      f->dump_format_unquoted("last", "%s", fixed_u_to_string(sitem.last,3).c_str());
       f->close_section(); // entry
     }
     f->close_section(); // entries
